src/mongoose_rabbit_worker.erl

```diff
@@ -0,0 +1,298 @@
+                         ?REPLY_REQ_DROPPED(State)).
+
+handle_cast(Req, State) ->
+   maybe_handle_request(fun do_handle_cast/2, [Req, State],
+                        ?NOREPLY_REQ_DROPPED(State)).
+
+handle_info(Req = {init, _}, State) ->
+    do_handle_info(Req, State);
+handle_info(Req, State) ->
+    maybe_handle_request(fun do_handle_info/2, [Req, State],
+                         ?NOREPLY_REQ_DROPPED(State)).
+
+terminate(_Reason, #{connection := Connection, channel := Channel,
+                     host := Host}) ->
+    close_rabbit_connection(Connection, Channel, Host),
+    ok.
+
+%%%===================================================================
+%%% Internal functions
+%%%===================================================================
+
+do_handle_call({amqp_call, Method}, _From, State = #{channel := Channel}) ->
+    try amqp_channel:call(Channel, Method) of
+        Res ->
+            {reply, {ok, Res}, State}
+    catch
+        Error:Reason ->
+            {FreshConn, FreshChann} = maybe_restart_rabbit_connection(State),
+            {reply, {Error, Reason}, State#{connection := FreshConn,
+                                            channel := FreshChann}}
+    end.
+
+do_handle_cast({amqp_publish, Method, Payload}, State) ->
+    handle_amqp_publish(Method, Payload, State).
+
+do_handle_info({init, Opts}, State) ->
+    Host = proplists:get_value(host, Opts),
+    AMQPClientOpts = proplists:get_value(amqp_client_opts, Opts),
+    {Connection, Channel} = establish_rabbit_connection(AMQPClientOpts, Host),
+    IsConfirmEnabled = maybe_enable_confirms(Channel, Opts),
+    MaxMsgQueueLen = proplists:get_value(max_queue_len, Opts),
+    {noreply, State#{host => Host, amqp_client_opts => AMQPClientOpts,
+                     connection => Connection, channel => Channel,
+                     confirms => IsConfirmEnabled,
+                     max_queue_len => MaxMsgQueueLen}}.
+
+-spec handle_amqp_publish(Method :: mongoose_amqp:method(),
+                          Payload :: mongoose_amqp:message(),
+                          Opts :: worker_opts()) -> {noreply, worker_opts()}.
+handle_amqp_publish(Method, Payload, Opts = #{host := Host}) ->
+    {PublishTime, Result} =
+        timer:tc(fun publish_message_and_wait_for_confirm/3,
+                 [Method, Payload, Opts]),
+    case Result of
+        true ->
+            update_messages_published_metrics(Host, PublishTime, Payload),
+            ?DEBUG("event=rabbit_message_sent message=~1000p", [Payload]),
+            {noreply, Opts};
+        false ->
+            update_messages_failed_metrics(Host),
+            ?WARNING_MSG("event=rabbit_message_sent_failed reason=negative_ack",
+                         []),
+            {noreply, Opts};
+        {channel_exception, Error, Reason} ->
+            update_messages_failed_metrics(Host),
+            ?WARNING_MSG("event=rabbit_message_sent_failed reason=~1000p:~1000p",
```

Comment:
So few details. No `Host`, no basic connection params that can be used for debugging... This applies to other logs in this file as well.